lib/services/ios-project-service.ts

```diff
@@ -169,6 +171,89 @@ class IOSProjectService implements  IPlatformProjectService {
+            }
+
+            var packageType = this.$childProcess.exec(util.format("/usr/libexec/PlistBuddy -c \"Print :CFBundlePackageType\" %s", infoPlistPath)).wait().trim();
+            if (packageType !== "FMWK") {
+                this.$errors.failWithoutHelp("The bundle at %s does not appear to be a dynamic framework.", libraryPath);
+            }
+        }).future<void>()();
+    }
+
+    private getUmbrellaHeaderFromDynamicFramework(libraryPath: string): IFuture<string> {
+        return (() => {
+            var modulemapPath = path.join(libraryPath, "Modules", "module.modulemap");
+            if (!this.$fs.exists(modulemapPath).wait()) {
+				this.$errors.failWithoutHelp("The framework at %s does not contain a module.modulemap file.", modulemapPath);
+            }
+
+            var modulemap = this.$fs.readText(modulemapPath).wait();
+            var umbrellaRegex = /umbrella header "(.+\.h)"/g;
+            var match = umbrellaRegex.exec(modulemap);
+            if (!match) {
+                this.$errors.failWithoutHelp("The modulemap at %s does not specify an umbrella header.", modulemapPath);
+            }
+
+            return match[1];
+        }).future<string>()();
+    }
+
+    private generateFrameworkMetadata(projectRoot: string, frameworkDir: string, frameworkName: string, umbrellaHeader: string): IFuture<void> {
+        return (() => {
+            if (!this.$fs.exists("/usr/local/lib/libmonoboehm-2.0.1.dylib").wait()) {
+                this.$errors.failWithoutHelp("NativeScript needs Mono 3.10 or newer installed in /usr/local");
+            }
+
+            var yamlOut = path.join(frameworkDir, "Metadata");
+            this.$fs.createDirectory(yamlOut).wait();
+
+            var tempHeader = path.join(yamlOut, "Metadata.h");
+            this.$fs.writeFile(tempHeader, util.format("#import <%s/%s>", frameworkName, umbrellaHeader)).wait();
+
+            this.$logger.info("Generating metadata for %s.framework. This can take a minute.", frameworkName);
+            var sdkPath = this.$childProcess.exec("xcrun -sdk iphoneos --show-sdk-path").wait().trim();
+            var generatorExecOptions = {
+                env: {
+                    DYLD_FALLBACK_LIBRARY_PATH: this.$childProcess.exec("xcode-select -p").wait().trim() + "/Toolchains/XcodeDefault.xctoolchain/usr/lib"
+                }
+            };
+            this.$childProcess.exec(util.format("%s/Metadata/MetadataGenerator -s %s -u %s -o %s -cflags=\"-F%s\"", projectRoot, sdkPath, tempHeader, yamlOut, frameworkDir), generatorExecOptions).wait();
+
+            this.$fs.copyFile(path.join(yamlOut, "Metadata-armv7", frameworkName + ".yaml"), path.join(projectRoot, "Metadata", "Metadata-armv7", frameworkName + ".yaml")).wait();
+            this.$fs.copyFile(path.join(yamlOut, "Metadata-arm64", frameworkName + ".yaml"), path.join(projectRoot, "Metadata", "Metadata-arm64", frameworkName + ".yaml")).wait();
+
```

Comment:
consider extracting common paths like frameworkName + ".yaml" for easier reading.